core/src/main/java/org/apache/accumulo/core/spi/balancer/TableLoadBalancer.java

```diff
@@ -188,6 +188,22 @@ public void getAssignments(AssignmentParameters params) {
     }
   }
 
+  @Override
+  public boolean needsReassignment(CurrentAssignment currentAssignment) {
+    var tableId = currentAssignment.getTablet().getTable();
+    String value = environment.getConfiguration(tableId).get(TABLE_ASSIGNMENT_GROUP_PROPERTY);
+    String expectedGroup = (value == null || StringUtils.isEmpty(value))
+        ? Constants.DEFAULT_RESOURCE_GROUP_NAME : value;
+
+    if (!expectedGroup.equals(currentAssignment.getResourceGroup())) {
+      // The tablet is not in the expected resource group, so it needs to be reassigned
+      return true;
+    }
+
+    // defer to the per table balancer
+    return getBalancerForTable(tableId).needsReassignment(currentAssignment);
```

Comment:
Calling down to the table balancer could possibly allow for a per table balancer that partitions tablets to groups of tservers.